### PR TITLE
more solar panels

### PR DESCRIPTION
**Pull request by Riley31415**
## Short Description
Added a basic, lv, mv, hv solar panels, changed recipes and outputs, removed useless output

## Additions/Changes/Removals
basic solar panel 10 j/s day 10k storage
lv solar panel 100 j/s day 25k storage
mv solar panel 200 j/s day 50k storage
hv solar panel 400j/s day 100k storage
Advanced requires hv panel in recipe
advanced, hybrid, ultimate outputs x10 since they require more stuff

## Related Issues
none
## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 1]
```diff
@@ -32,6 +32,10 @@
     private static final CustomItem generatingItem = new CustomItem(Material.ORANGE_STAINED_GLASS_PANE,
         "&cNot Generating..."
     );
+    public static int BASIC_DAY_RATE = 10;
+    public static int BASIC_NIGHT_RATE = 1;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: 10 / 1 = 8 when?

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
You shoukd add lv mv and hv solar panels to then you can scale up the production of the advanced

**Comment by J3fftw1** [comment]
Also are you editing this via github?

**Comment by NCBPFluffyBear** [comment]
is the output parameter even used? cant we just cut it out? afaik its just day time and night time

**Comment by Riley31415** [comment]
made changes

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 2]
```diff
@@ -153,7 +193,7 @@ public int getCapacity() {
                 Items.CARBON_PLATE, new ItemStack(Material.LAPIS_BLOCK), Items.CARBON_PLATE,
                 Items.IRIDIUM_PLATE, Items.ADVANCED_MACHINE_BLOCK, Items.IRIDIUM_PLATE,
                 SlimefunItems.ADVANCED_CIRCUIT_BOARD, Items.IRIDIUM_PLATE, SlimefunItems.ADVANCED_CIRCUIT_BOARD
-            }),
+        }),
```
Comment (Riley31415) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: dunno wut i did here will it mess stuff up?

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: Please fix

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 3]
```diff
@@ -171,6 +171,10 @@ private void registerCarbonStuff() {
     }
 
     private void registerSolarPanels() {
+        new AdvancedSolarPanel(AdvancedSolarPanel.Type.BASIC).register(plugin);
+	    new AdvancedSolarPanel(AdvancedSolarPanel.Type.LV).register(plugin);
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Please fix

File: src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java [hunk 4]
```diff
@@ -100,7 +100,7 @@ private void setupResearches() {
 
         new Research(new NamespacedKey(this, "the_better_panel"),
             696973, "These are the better panels", 45)
-            .addItems(Items.ADVANCED_SOLAR_PANEL, Items.ULTIMATE_SOLAR_PANEL, Items.HYBRID_SOLAR_PANEL)
+            .addItems(Items.BASIC_SOLAR_PANEL, Items.LV_SOLAR_PANEL, Items.MV_SOLAR_PANEL, Items.HV_SOLAR_PANEL, Items.ADVANCED_SOLAR_PANEL, Items.ULTIMATE_SOLAR_PANEL, Items.HYBRID_SOLAR_PANEL)
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java]: Dont make such long line

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 5]
```diff
@@ -171,6 +171,10 @@ private void registerCarbonStuff() {
     }
 
     private void registerSolarPanels() {
+        new AdvancedSolarPanel(AdvancedSolarPanel.Type.BASIC).register(plugin);
+	new AdvancedSolarPanel(AdvancedSolarPanel.Type.LV).register(plugin);
+        new AdvancedSolarPanel(AdvancedSolarPanel.Type.MV).register(plugin);
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Give them better names not mv lv or hv do the actual name

File: src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java [hunk 6]
```diff
@@ -100,6 +100,8 @@ private void setupResearches() {
 
         new Research(new NamespacedKey(this, "the_better_panel"),
             696973, "These are the better panels", 45)
+            .addItems(Items.BASIC_SOLAR_PANEL)
+            .addItems(Items.LV_SOLAR_PANEL, Items.MV_SOLAR_PANEL, Items.HV_SOLAR_PANEL)
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java]: What the hell?

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 7]
```diff
@@ -32,16 +32,32 @@
...
+    public static int HV_NIGHT_RATE = 50;
+    public static int HV_OUTPUT = 400;
+    public static int HV_STORAGE = 100_000;
+    public static int ADVANCED_DAY_RATE = 800;
+    public static int ADVANCED_NIGHT_RATE = 100;
     public static int ADVANCED_OUTPUT = 320;
     public static int ADVANCED_STORAGE = 320_000;
-    public static int HYBRID_DAY_RATE = 640;
-    public static int HYBRID_NIGHT_RATE = 80;
+    public static int HYBRID_DAY_RATE = 6400;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: 6_400

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 8]
```diff
@@ -32,16 +32,32 @@
...
+    public static int HYBRID_NIGHT_RATE = 800;
     public static int HYBRID_OUTPUT = 1200;
     public static int HYBRID_STORAGE = 1_000_000;
-    public static int ULTIMATE_DAY_RATE = 5120;
-    public static int ULTIMATE_NIGHT_RATE = 640;
+    public static int ULTIMATE_DAY_RATE = 51200;
+    public static int ULTIMATE_NIGHT_RATE = 6400;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: 51_200
6_400

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 9]
```diff
@@ -32,16 +32,32 @@
...
+    public static int HV_STORAGE = 100_000;
+    public static int ADVANCED_DAY_RATE = 800;
+    public static int ADVANCED_NIGHT_RATE = 100;
     public static int ADVANCED_OUTPUT = 320;
     public static int ADVANCED_STORAGE = 320_000;
-    public static int HYBRID_DAY_RATE = 640;
-    public static int HYBRID_NIGHT_RATE = 80;
+    public static int HYBRID_DAY_RATE = 6400;
+    public static int HYBRID_NIGHT_RATE = 800;
     public static int HYBRID_OUTPUT = 1200;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: 1_200

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java (hunk 6 above)
Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java]: idfk u said not to have long lines, u want me to do another research instead? in that case wut numbers do i put

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java (hunk 5 above)
Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: wut are there real names lul

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Low voltage hight voltage medium voltage

File: src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java (hunk 6 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java]: You can also press enter on a line

Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java]: oooo

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 10]
```diff
@@ -32,17 +32,33 @@
...
+    public static int ADVANCED_DAY_RATE = 800;
+    public static int ADVANCED_NIGHT_RATE = 100;
+    public static int ADVANCED_OUTPUT = 0;
     public static int ADVANCED_STORAGE = 320_000;
-    public static int HYBRID_DAY_RATE = 640;
-    public static int HYBRID_NIGHT_RATE = 80;
-    public static int HYBRID_OUTPUT = 1200;
+    public static int HYBRID_DAY_RATE = 6_400;
+    public static int HYBRID_NIGHT_RATE = 800;
+    public static int HYBRID_OUTPUT = 0;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: They dont output energy anymore

Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: thats the one u dont use

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: Whut?

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: Who told you that

Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: the OUTPUT isnt used anywhere u only use day rate and night rate

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: If its not used anywhere why dont you remove it

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 11]
```diff
@@ -172,9 +172,9 @@ private void registerCarbonStuff() {
 
     private void registerSolarPanels() {
         new AdvancedSolarPanel(AdvancedSolarPanel.Type.BASIC).register(plugin);
-	new AdvancedSolarPanel(AdvancedSolarPanel.Type.LV).register(plugin);
-        new AdvancedSolarPanel(AdvancedSolarPanel.Type.MV).register(plugin);
-        new AdvancedSolarPanel(AdvancedSolarPanel.Type.HV).register(plugin);
+	    new AdvancedSolarPanel(AdvancedSolarPanel.Type.LOW_VOLTAGE).register(plugin);
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Formatting

File: src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java [hunk 12]
```diff
@@ -100,7 +100,8 @@ private void setupResearches() {
 
         new Research(new NamespacedKey(this, "the_better_panel"),
             696973, "These are the better panels", 45)
-            .addItems(Items.ADVANCED_SOLAR_PANEL, Items.ULTIMATE_SOLAR_PANEL, Items.HYBRID_SOLAR_PANEL)
+            .addItems(Items.BASIC_SOLAR_PANEL, Items.LOW_VOLTAGE_SOLAR_PANEL, Items.MEDIUM_VOLTAGE_SOLAR_PANEL, Items.HIGH_VOLTAGE_SOLAR_PANEL,
+                      Items.ADVANCED_SOLAR_PANEL, Items.ULTIMATE_SOLAR_PANEL, Items.HYBRID_SOLAR_PANEL)
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/LiteXpansion.java]: Closing bracked new line

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

**Comment by J3fftw1** [comment]
Build failed you should really test this stuff

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 13]
```diff
@@ -32,17 +32,26 @@
...
-    public static int HYBRID_OUTPUT = 1200;
+    public static int HYBRID_DAY_RATE = 6_400;
+    public static int HYBRID_NIGHT_RATE = 800;
     public static int HYBRID_STORAGE = 1_000_000;
-    public static int ULTIMATE_DAY_RATE = 5120;
-    public static int ULTIMATE_NIGHT_RATE = 640;
-    public static int ULTIMATE_OUTPUT = 5120;
+    public static int ULTIMATE_DAY_RATE = 51_200;
+    public static int ULTIMATE_NIGHT_RATE = 6_400;
     public static int ULTIMATE_STORAGE = 10_000_000;
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: This is getting quite unorganized, you already have an enum for the different types, so you should rather add these values to the corresponding enum as fields instead.

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 14]
```diff
@@ -171,6 +171,10 @@ private void registerCarbonStuff() {
     }
 
     private void registerSolarPanels() {
+        new AdvancedSolarPanel(AdvancedSolarPanel.Type.BASIC).register(plugin);
+	    new AdvancedSolarPanel(AdvancedSolarPanel.Type.LOW_VOLTAGE).register(plugin);
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: indentation

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java (hunk 13 above)
Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: i got no idea wut that means ¯\_(ツ)_/¯

**Comment by Riley31415** [comment]
Tested it all seems to be working

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 15]
```diff
@@ -141,22 +150,39 @@ public int getCapacity() {
     @AllArgsConstructor(access = AccessLevel.PRIVATE)
     public enum Type {
 
-        ADVANCED(Items.ADVANCED_SOLAR_PANEL, ADVANCED_DAY_RATE, ADVANCED_NIGHT_RATE, ADVANCED_OUTPUT,
-            ADVANCED_STORAGE, new ItemStack[] {
+        BASIC(Items.BASIC_SOLAR_PANEL, 10, 1, 10_000, new ItemStack[] {
```
Comment (Riley31415) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: Is this what you mean by 'you should rather add these values to the corresponding enum as fields instead'? @TheBusyBiscuit

File: src/main/java/dev/j3fftw/litexpansion/Items.java [hunk 16]
```diff
@@ -272,6 +272,46 @@
...
+        "HIGH_VOLTAGE_SOLAR_PANEL",
+        Material.YELLOW_GLAZED_TERRACOTTA,
+        "&e&lHigh Voltage Solar Panel",
+        "&9Works at Night",
+        "",
+        LoreBuilderDynamic.powerBuffer(AdvancedSolarPanel.HIGH_VOLTAGE_STORAGE),
+        LoreBuilderDynamic.powerPerTick(AdvancedSolarPanel.HIGH_VOLTAGE_DAY_RATE) + " (Day)",
+        LoreBuilderDynamic.powerPerTick(AdvancedSolarPanel.HIGH_VOLTAGE_NIGHT_RATE) + " (Night)"
+    );
     public static final SlimefunItemStack ADVANCED_SOLAR_PANEL = new SlimefunItemStack(
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Items.java:307]: New ones shouldnt work at night

Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/Items.java:307]: Then they wont work at all cuz of the only nightime rate bug

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Items.java:307]: Then that issue should be fixed. I still dont want these to work at night. No matter the issue

**Comment by J3fftw1** [comment]
Remove it from the enum if its not needed

**Comment by Riley31415** [comment]
What is it?

**Comment by J3fftw1** [comment]
Night time sorry thought i commented on chanfe

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
Its still in the code remoce it completely from these machines

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 17]
```diff
@@ -150,22 +146,22 @@ public int getCapacity() {
     @AllArgsConstructor(access = AccessLevel.PRIVATE)
     public enum Type {
 
-        BASIC(Items.BASIC_SOLAR_PANEL, 10, 1, 10_000, new ItemStack[] {
+        BASIC(Items.BASIC_SOLAR_PANEL, BASIC_DAY_RATE, 0, BASIC_STORAGE, new ItemStack[] {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java:149]: Remove the 0

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 18]
```diff
@@ -141,22 +146,39 @@ public int getCapacity() {
...
 
-        ADVANCED(Items.ADVANCED_SOLAR_PANEL, ADVANCED_DAY_RATE, ADVANCED_NIGHT_RATE, ADVANCED_OUTPUT,
-            ADVANCED_STORAGE, new ItemStack[] {
+        BASIC(Items.BASIC_SOLAR_PANEL, BASIC_DAY_RATE, 0, BASIC_STORAGE, new ItemStack[] {
+            Items.CARBON_PLATE, Items.CARBON_PLATE, Items.CARBON_PLATE,
             Items.REINFORCED_GLASS, Items.REINFORCED_GLASS, Items.REINFORCED_GLASS,
-            Items.ADVANCED_ALLOY, SlimefunItems.SOLAR_GENERATOR_4, Items.ADVANCED_ALLOY,
+            Items.ELECTRONIC_CIRCUIT, Items.MACHINE_BLOCK, Items.ELECTRONIC_CIRCUIT
+        }),
+	    LOW_VOLTAGE(Items.LOW_VOLTAGE_SOLAR_PANEL, LOW_VOLTAGE_DAY_RATE, 0, LOW_VOLTAGE_STORAGE, new ItemStack[] {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java:154]: Remove the 0

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 19]
```diff
@@ -141,22 +146,39 @@ public int getCapacity() {
...
+            Items.REINFORCED_GLASS, Items.REINFORCED_GLASS, Items.REINFORCED_GLASS,
+            Items.ADVANCED_ALLOY, Items.BASIC_SOLAR_PANEL, Items.ADVANCED_ALLOY,
+            Items.ADVANCED_CIRCUIT, Items.ADVANCED_MACHINE_BLOCK, Items.ADVANCED_CIRCUIT
+        }),
+	    MEDIUM_VOLTAGE(Items.MEDIUM_VOLTAGE_SOLAR_PANEL, MEDIUM_VOLTAGE_DAY_RATE, 0, MEDIUM_VOLTAGE_STORAGE, new ItemStack[] {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java:159]: Remove the 0

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 20]
```diff
@@ -141,22 +146,39 @@ public int getCapacity() {
...
+            Items.REINFORCED_GLASS, Items.REINFORCED_GLASS, Items.REINFORCED_GLASS,
+            Items.ADVANCED_ALLOY, Items.LOW_VOLTAGE_SOLAR_PANEL, Items.ADVANCED_ALLOY,
+            Items.ADVANCED_CIRCUIT, Items.ADVANCED_MACHINE_BLOCK, Items.ADVANCED_CIRCUIT
+        }),
+	    HIGH_VOLTAGE(Items.HIGH_VOLTAGE_SOLAR_PANEL, HIGH_VOLTAGE_DAY_RATE, 0, HIGH_VOLTAGE_STORAGE, new ItemStack[] {
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java:164]: Remove the 0

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java (hunk 20 above)
Comment (Riley31415) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java:164]: Wont that break the thing below it? its looking for 4 things and will only get 3

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java:164]: Make a new 1 that accepts 3

**Pull request closed without merging**